website/notifications/events/files.py

```diff
@@ -0,0 +1,315 @@
+"""File event module
+
+Contains usable classes:
+- FileAdded
+- FileUpdated
+- FileRemoved
+- FolderCreated
+- AddonFileMoved
+- AddonFileCopied
+
+These classes are registered in event_register and are callable through the
+ register. The main way these are used is with the signals blinker module
+ that catches the signal with the addon_file_updated function.
+
+FileEvent and ComplexFileEvent are parent classes with shared functionality.
+"""
+from furl import furl
+
+from website.notifications import emails
+from website.notifications.events.model import register, Event, event_register
+from website.notifications.constants import NOTIFICATION_TYPES
+from website.models import Node
+from website.project.model import NodeLog
+from website.notifications.events import utils as event_utils
+from website.notifications.events.signals import file_updated as signal
+from website.notifications import utils
+
+
+@signal.connect
+def file_updated(self, user=None, node=None, event_type=None, payload=None):
+    if event_type not in event_register:
+        raise TypeError
```

Comment:
`TypeError` doesn't seem correct here (we aren't dealing with programming "types" here); maybe raise a custom `RegistryError`.